packages/web3-shared/base/src/utils/fetch.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Fetch image by creating an image element.
+ * It's not supporting in mv3 on background page.
+ * @param url
+ * @returns
+ */
+export function fetchImageByDOM(url: string) {
+    return new Promise<Blob | null>((resolve, reject) => {
+        const img = document.createElement('img')
+        const cleanup = () => {
+            img.removeEventListener('load', onload)
+            img.removeEventListener('error', reject)
+        }
+        const onload = () => {
+            const canvas = document.createElement('canvas')
+            const context = canvas.getContext('2d')
+            canvas.height = img.naturalHeight
```

Comment:
Use https://developer.mozilla.org/zh-CN/docs/Web/API/OffscreenCanvas instead.

Reply:
It's an experiment API. I'm hesitant to use it.